bblfsh/python-driver
# [INFORMATIVE, NOT FOR MERGE] Simple pypy test and results

How to test:

- Get this branch.
- `sh run_bench.sh`

It should build and image with pypy, run the benchmark (parsing all the files in the Python stdlib), then change the native driver entry point (in a ghetto way), restart the container and  run the same benchmarks with Python 3. Times are printed for every run of the benchmark.

This uses a single driver instance, without the server, and do the requests serially, since we want to test the native driver speed. But the results with the server are similar.

On my machine, times with **Python 3 are in the 30-36 seconds** range and with **pypy in the 48-59** range. 

So it looks like pypy doesn't works pretty well here, even tough the native driver should be kept running which in theory would benefit the JIT allowing it to warm up. On the other side, this matches my previous experiences with pypy, which only usually give a big boost on math related things.

I didn't check exactly were the time differences are, but it would be simple to add if there is interest.
